pkg/tidb/client.go

```diff
@@ -66,22 +67,28 @@ func NewTiDBClient(lc fx.Lifecycle, config *config.Config, etcdClient *clientv3.
 		_ = mysql.RegisterTLSConfig(sqlAPITLSKey, config.TiDBTLSConfig)
 	}
 
+	cache := ttlcache.NewCache()
+	cache.SkipTTLExtensionOnHit(true)
 	client := &Client{
 		lifecycleCtx:             nil,
 		forwarder:                newForwarder(lc, etcdClient),
+		endpointAllowlist:        []string{},
 		statusAPIHTTPScheme:      config.GetClusterHTTPScheme(),
 		statusAPIAddress:         "",
 		enforceStatusAPIAddresss: false,
 		statusAPIHTTPClient:      httpClient,
 		statusAPITimeout:         defaultTiDBStatusAPITimeout,
 		sqlAPITLSKey:             sqlAPITLSKey,
 		sqlAPIAddress:            "",
-		cache:                    ttlcache.NewCache(),
+		cache:                    cache,
 	}
 
 	lc.Append(fx.Hook{
 		OnStart: func(ctx context.Context) error {
 			client.lifecycleCtx = ctx
+			statusProxyIP, statusProxyPort, _ := host.ParseHostAndPortFromAddressURL("http://" + client.forwarder.statusProxy.listener.Addr().String())
```

Comment:
Will this cause problems under TLS?

Reply:
Nope, but I found it unnecessary to parse the `net.Addr`.